chapters/VK_EXT_present_timing/queries.txt

```diff
@@ -0,0 +1,441 @@
+presented, and can: determine when an image was actually presented.
+This can allow the application's animation to look smooth to the user, with
+no stuttering.
+The
+ifdef::VK_EXT_present_timing+VK_GOOGLE_display_timing[]
+`VK_EXT_present_timing` and `VK_GOOGLE_display_timing` extensions allow
+endif::VK_EXT_present_timing+VK_GOOGLE_display_timing[]
+ifdef::VK_EXT_present_timing[]
+`VK_EXT_present_timing` extension allows
+endif::VK_EXT_present_timing[]
+ifdef::VK_GOOGLE_display_timing[]
+`VK_GOOGLE_display_timing` extension allows
+endif::VK_GOOGLE_display_timing[]
+an application to satisfy these needs.
+
+The presentation engine's display typically refreshes the pixels that are
+displayed to the user on a periodic basis.
+The period may be fixed or variable.
+In many cases, the presentation engine is associated with fixed refresh rate
+(FRR) display technology, with a fixed refresh rate (RR, e.g. 60Hz).
+In some cases, the presentation engine is associated with variable refresh
+rate (VRR) display technology, where each refresh cycle (RC) can vary in
+length.
+
+ifdef::VK_EXT_present_timing[]
+[open,refpage='vkGetSwapchainTimingPropertiesEXT',desc='Obtain the display timing properties of the PE\'s display',type='protos']
+--
+
+To query the presentation engine's timing properties for a given swapchain,
+call:
+
+include::{generated}/api/protos/vkGetSwapchainTimingPropertiesEXT.txt[]
+
+  * pname:device is the device associated with pname:swapchain.
+  * pname:swapchain is the swapchain to obtain timing properties for.
+  * pname:pSwapchainTimingProperties is a pointer to an instance of the
+    sname:VkSwapchainTimingPropertiesEXT structure.
+
+include::{generated}/validity/protos/vkGetSwapchainTimingPropertiesEXT.txt[]
+--
+
+[open,refpage='VkSwapchainTimingPropertiesEXT',desc='Structure containing the RC duration of a display',type='structs']
+--
+
+The sname:VkSwapchainTimingPropertiesEXT structure is defined as:
+
+include::{generated}/api/structs/VkSwapchainTimingPropertiesEXT.txt[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to an extension-specific structure.
+  * pname:refreshDuration is zero; or is an indication of the duration
+    of a refresh cycle.
+    If the presentation engine is operating as an FRR display, this is the
+    number of nanoseconds from the start of one refresh cycle to the
+    start of the next refresh cycle.
+    If the presentation engine is operating as an VRR display
+    (i.e. refresh cycles may: have variable length), this is the
+    minimum number of nanoseconds from the start of one refresh cycle
+    to the start of the next refresh cycle.
+  * pname:variableRefresh is undefined: if pname:refreshDuration is
+    zero; otherwise it is ename:VK_FALSE if the presentation engine is
+    operating as a FRR display, or ename:VK_TRUE if the presentation
+    engine is operating as a VRR display.
+
```

Comment:
Would this be a good place to also report if the swapchain's associated presentation engine is _in principle_ capable of VRR, via a VkBool variableRefreshSupported? I'd like to have a way to find out if VRR is in principle supported on a swapchain's display (ie. driver + gpu + display support it and it is enabled by compositor/user). "variableRefresh" describes if VRR mode is active at time of query.

E.g., if connected display or gpu or display server/driver is incapable of VRR or if VRR is disabled by the user in some display setting, then both variableRefreshSupported and variableRefresh would be VK_FALSE. But if the setup is capable of VRR and just at this specific time of query VRR mode is not active, maybe because the window is not a fullscreen window or partially obscured, then variableRefreshSuppported would be VK_TRUE, but variableRefresh would be VK_FALSE.

This would be very useful for research applications but presumably also games and other apps to provide users with more accurate feedback/warnings or some troubleshooting instructions if the given use case requires VRR and VRR is not active, e.g., to figure out if it is because of obscured window / desktop composition kicking in, or instead because of unsuitable display/gpu or user error (=user disabled the required VRR feature in some display setting GUI or config file). My toolkit does this on Linux/X11, involving lots of non-portable hacks, but a proper and portable reporting mechanism in this extension would be splendid.